### PR TITLE
Can criticism indices

### LINKED ISSUE #2340: Add indices to scvi criticism if using other subset AnnData object
Add indices here.

https://github.com/scverse/scvi-tools/blob/021fc36a75fc0b825f306133eba908cc7c6b2215/scvi/criticism/_ppc.py#L94C49-L94C49

### PR DESCRIPTION
**Pull request by canergen**
-   [X ] Closes #2340 
-   [X] Tests added and passed if fixing a bug or adding a new feature
-   [X] All code checks passed
-   [X] Added type annotations to new arguments/methods/functions
-   [x] Added an entry in the latest `docs/release_notes/index.md` file if fixing a bug or adding a new feature
-   [x] If the changes are patches for a version, I have added the `on-merge: backport to x.x.x` label

### CODE REVIEW

**Comment by canergen** [comment]
@martinkim0 Fixed indices in criticism.

**Comment by martinkim0** [comment]
Just needs a `from __future__ import annotations` at the top of the file.

File: tests/criticism/test_criticism.py [hunk 1]
```diff
@@ -27,20 +28,20 @@ def get_ppc_with_samples(adata: AnnData, n_samples: int = 2):
...
     models_dict = {"model1": model1, "model2": model2}
-    ppc = PPC(adata, models_dict, n_samples=n_samples)
+    ppc = PPC(adata, models_dict, n_samples=n_samples, indices=indices)
     return ppc, models_dict
 
 
 def test_ppc_init():
     adata = synthetic_iid()
-    ppc, models_dict = get_ppc_with_samples(adata, n_samples=42)
+    ppc, models_dict = get_ppc_with_samples(adata, n_samples=42, indices=np.arange(100))
```
Comment (martinkim0) [inline tests/criticism/test_criticism.py:37]: We should also keep a call to `get_ppc_with_samples` without passing in `indices`.

**Review by martinkim0 — COMMENTED** [review]
LGMT, one minor comment and needs a release note

**Comment by martinkim0** [comment]
I'm still confused as to whether criticism metrics should use `model.adata` or the user-inputted `adata`. I feel like it should default to `model.adata` if `adata` is not given, but otherwise should use `adata` for all computations after validating that it is compatible with the model.

**Comment by canergen** [comment]
They should use model.adata. The inputted adata should be model.adata. However, we need the adata argument in case model.adata is minified in which case it doesn't contain raw counts.

**Comment by canergen** [comment]
@watiss You were opposed about a default to model.adata. We should then check that model.adata is not minified, in which case the default doesn't make sense.

File: docs/release_notes/index.md [hunk 2]
```diff
@@ -57,6 +57,8 @@ is available in the [commit logs](https://github.com/scverse/scvi-tools/commits/
 -   Fix bug in {class}`scvi.module.SCANVAE` where classifier probabilities
     were interpreted as logits. This is backwards compatible as loading older
     models will use the old code path {pr}`2301`.
+-   scvi.criticism allows running metrics on a subset of the full adata set
+    by also subsetting model.adata.
```
[the remark this answers is not in the document]
Comment (martinkim0) [reply docs/release_notes/index.md:61]: Did you mean to resolve this?

**Review by martinkim0 — APPROVED** [review]
